dask/dataframe/io/parquet/core.py

```diff
@@ -553,31 +553,55 @@ def to_parquet(
     # Use i_offset and df.npartitions to define file-name list
     filenames = ["part.%i.parquet" % (i + i_offset) for i in range(df.npartitions)]
 
-    # write parts
-    dwrite = delayed(engine.write_partition)
-    parts = [
-        dwrite(
-            d,
+    # Construct IO graph
+    dsk = {}
+    name = "to-parquet-" + tokenize(
+        df,
+        fs,
+        path,
+        append,
+        ignore_divisions,
+        partition_on,
+        division_info,
+        index_cols,
+        schema,
+    )
+    part_tasks = []
+    for d, filename in enumerate(filenames):
+        kwargs_pass["fmd"] = meta
+        kwargs_pass["compression"] = compression
+        kwargs_pass["index_cols"] = index_cols
+        kwargs_pass["schema"] = schema
```

Comment:
These updates on `kwargs_pass` can be done outside of the for loop.

Reply:
Oops!